podvm/qcow2/misc-settings.sh

```diff
@@ -38,85 +38,37 @@ then
     esac
 fi
 
-if [[ "$CLOUD_PROVIDER" == "azure" && "$PODVM_DISTRO" == "ubuntu" ]]; then
+if [[ "$CLOUD_PROVIDER" == "azure" || "$CLOUD_PROVIDER" == "generic" ]] && [[ "$PODVM_DISTRO" == "ubuntu" ]]; then
     export DEBIAN_FRONTEND=noninteractive
+    curl -fsSL https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
+    echo "deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+
```

Comment:
I think this `podvm/qcow2/misc-settings.sh` is also used when s390x images are built with Packer.

@liudalibj @huoqifeng Is this correct?

Reply:
yeah, the script be used at here https://github.com/confidential-containers/cloud-api-adaptor/blob/main/podvm/qcow2/ubuntu/qemu-ubuntu.pkr.hcl#L86
when s390x images are built with packer.

Reply:
I won't be installed as CLOUD_PROVIDER=ibmcloud in this case, no?

Reply:
@yoheiueda @liudalibj should i remove it or this is ok?

Reply:
@snir911 yeah, this is OK.

Reply:
it's ok here, I did a test to build one s390x image with this PR, the related logs
```
==> qemu.ubuntu: Uploading qcow2/misc-settings.sh => ~/misc-settings.sh
==> qemu.ubuntu: Provisioning with shell script: /tmp/packer-shell122513875
    qemu.ubuntu: passwd: password expiry information changed.
==> qemu.ubuntu: Removed /etc/systemd/system/timers.target.wants/apt-daily.timer.
==> qemu.ubuntu: Removed /etc/systemd/system/timers.target.wants/apt-daily-upgrade.timer.
==> qemu.ubuntu: Removed /etc/systemd/system/multi-user.target.wants/snapd.service.
==> qemu.ubuntu: Removed /etc/systemd/system/multi-user.target.wants/snapd.seeded.service.
==> qemu.ubuntu: Removed /etc/systemd/system/cloud-final.service.wants/snapd.seeded.service.
==> qemu.ubuntu: Removed /etc/systemd/system/multi-user.target.wants/snap.lxd.activate.service.
==> qemu.ubuntu: Uploading qcow2/build-s390x-se-image.sh => ~/build-s390x-se-image.sh
```

Reply:
And when building one s390x podvm image the `CLOUD_PROVIDER` is empty inside the podvm_builder container.